packages/node/src/indexer/api.wrapper.ts

```diff
@@ -0,0 +1,217 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { ApiPromise } from '@polkadot/api';
+import { ApiInterfaceEvents, ApiOptions } from '@polkadot/api/types';
+import {
+  BlockHash,
+  SignedBlock,
+  RuntimeVersion,
+  Header,
+} from '@polkadot/types/interfaces';
+
+export class ApiWrapper {
```

Comment:
I think it would be good to extract an interface from this and have an implementation for Polkadot and for Algorand.